packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/apis/find.ts

```diff
@@ -230,57 +228,67 @@ export const performFind = async <T = unknown, A = unknown>(
     return SavedObjectsUtils.createEmptyFindResponse<T, A>(options);
   }
 
-  let result: SavedObjectsFindResponse<T, A>;
+  // if extensions are enabled, we need to build an updated authorization type map
+  // to pass on to the redact method
+  const redactTypeMapParams: GetFindRedactTypeMapParams | undefined = disableExtensions
+    ? undefined
+    : {
+        previouslyCheckedNamespaces: spacesToAuthorize,
+        objects: [],
+      };
+  const savedObjects: SavedObjectsFindResult[] = [];
+
+  body.hits.hits.forEach((hit: estypes.SearchHit<SavedObjectsRawDocSource>) => {
+    const obj = serializerHelper.rawToSavedObject(hit as SavedObjectsRawDoc, {
+      migrationVersionCompatibility,
+    });
+
+    if (redactTypeMapParams) {
+      redactTypeMapParams.objects.push({
+        type: obj.type,
+        id: obj.id,
+        existingNamespaces: obj.namespaces ?? [],
+      });
+    }
```

Comment:
I had to refactor a bit the internals of `find`, as we were creating the SO docs, migrating them, building the encryption params and then decrypting them, and we now need to have the encryption params built before calling `migrateAndDecryptStorageDocument` for each doc.